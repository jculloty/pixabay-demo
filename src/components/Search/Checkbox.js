import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Label from './Label';

class Checkbox extends PureComponent {
  static propTypes = {
    config: PropTypes.object.isRequired,
    checked: PropTypes.bool.isRequired,
    handleChange: PropTypes.func.isRequired,
  };

  handleChange = (event) => {
    const { name } = this.props.config;
    this.props.handleChange(name, event.target.checked ? 'true' : 'false');
  }

  render() {
    const { name } = this.props.config;

    return (
      <div>
        <Label name={name} />
        <input type="checkbox" id={name} checked={this.props.checked} onChange={this.handleChange} />
      </div>
    );
  }
}

export default Checkbox;
