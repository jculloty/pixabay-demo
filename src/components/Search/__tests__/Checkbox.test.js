import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'test-utils/setup-enzyme';
import Checkbox from '../Checkbox';

const buildComponent = function(props = {}) {
  props = {
    ...{
      config: { name: 'testing' },
      checked: true,
      handleChange: () => {},
    },
    ...props,
  };

  return (<Checkbox {...props} />);
};

describe('Search: Checkbox', () => {
  it('renders correctly', () => {
    let tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer
      .create(buildComponent({ checked: false }))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checking/unchecking triggers the correct handler', () => {
    const mockChange = jest.fn();
    const wrapper = shallow(buildComponent({ handleChange: mockChange }));

    wrapper.find('input').simulate('change', { target: { checked: false } });

    expect(mockChange).toHaveBeenCalledTimes(1);
    expect(mockChange).toHaveBeenCalledWith('testing', 'false');

    wrapper.find('input').simulate('change', { target: { checked: true } });
    expect(mockChange).toHaveBeenCalledWith('testing', 'true');
  });
});
