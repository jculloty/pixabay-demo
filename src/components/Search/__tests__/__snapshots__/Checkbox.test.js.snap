// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: Checkbox renders correctly 1`] = `
<div>
  <label
    htmlFor="testing"
  >
    Testing
  </label>
  <input
    checked={true}
    id="testing"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`Search: Checkbox renders correctly 2`] = `
<div>
  <label
    htmlFor="testing"
  >
    Testing
  </label>
  <input
    checked={false}
    id="testing"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
